.github/workflows/build.yml

```diff
@@ -0,0 +1,56 @@
+
+name: build
+on: [push]
+
+jobs:
+
+  install:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 20
+          cache: 'npm'
+
+      - uses: actions/cache@v4
+        env:
+          cache-name: cache-node-modules
+        with:
+          # npm cache files are stored in `~/.npm` on Linux/macOS
+          path: ~/.npm
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+            
+      - run: npm ci
+      # - run: npm run lint
+      - run: npm run build
+      - run: npm run test
+
+  semantic-release:
+    needs: install
+    runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/main'
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 20
+          cache: 'npm'
+      - run: npm ci
+      - run: npm run build
+      - name: Semantic Release
+        uses: cycjimmy/semantic-release-action@v4
+        env:
+          GITHUB_TOKEN: ${{ secrets.SR_GITHUB_TOKEN }}
```

Comment:
@spetz  semantic release needs a classic github token with "repo" scope from an user able to push on main (it need to bypass branch protection to update package.json's version and write CHANGELOG.md after a PR is merged ) the secret's name SR_GITHUB_TOKEN

Reply:
you can see how it ran on my own repo before here https://github.com/T1B0/iggy-bin/actions/runs/9763498796/job/26949566223